crates/metadata/src/tests.rs

```diff
@@ -169,3 +169,32 @@ fn spec_contract_json() {
         })
     )
 }
+
+#[test]
+fn trim_docs() {
+    // given
+    let name = "foo";
+    let cs = ConstructorSpec::from_name(name)
+        .selector(123_456_789u32.to_be_bytes())
+        .docs(vec![" foobar      "])
```

Comment:
Maybe this would be better if it was "" foo bar      ", because this test would also pass with just removing all spaces

Reply:
That's a good point, I'll include this change in another PR.